### PR TITLE
Extended radio api and AT86RF231 implementation

### PR DESCRIPTION
**Pull request by rousselk**
This PR completes the implementation of the AT86RF231 radio driver; the latter now implements all of the <code>netdev_802154_driver_t</code> features (that were until now only implemented as stubs), and adds some new options to this interface:
* <code>NETDEV_OPT_CAN_MONITOR</code> indicates whether the transceiver can go into "monitor" (a.k.a.: "promiscuous") mode;
* <code>NETDEV_OPT_CCA_THRESHOLD</code> for getting or setting the threshold value that is used by a transceiver to determine if radio channel is busy (CCA);
* <code>NETDEV_OPT_SFD_COUNT</code> for getting and resetting the number of SFDs (start of frame delimiters) detected by the transceiver: I think it is a good indicator to help estimate the traffic on the radio medium;
The two last options are planned to be used in future MAC protocol implementations.

### CODE REVIEW

**Comment by rousselk** [comment]
Note that this enhanced driver has been tested with success on FIT IoT-LAB's M3 nodes.

**Comment by gebart** [comment]
How do I test this? Can I use the rpl_udp example? Do I need any particular commands in the rpl shell?

**Comment by rousselk** [comment]
A new 'sfdcnt' shell command has been added in the shell (<code>sys/shell/commands/sc_transceiver.c</code>, see last commit).

File: drivers/at86rf231/at86rf231.c [hunk 1]
```diff
@@ -25,11 +26,12 @@
 #include "periph/gpio.h"
 #include "periph/spi.h"
 #include "kernel_types.h"
+#include "crash.h"
 #include "transceiver.h"
 #include "hwtimer.h"
 #include "config.h"
 
-#define ENABLE_DEBUG (0)
+#define ENABLE_DEBUG (1)
```
Comment (OlegHahm) [inline drivers/at86rf231/at86rf231.c]: should stay disabled by default

**Comment by OlegHahm** [comment]
Since @thomaseichinger is woring on the driver, too, I guess it makes sense if he reviews.

**Comment by PeterKietzmann** [comment]
@rousselk @haukepetersen @bapclenet does it make sense to keep this PR open?

**Comment by PeterKietzmann** [comment]
(just because of #2825)

**Comment by haukepetersen** [comment]
For some reason I was un-aware of this PR. As far as I see it, it contains some functionality that is not included in the remodeled ng_at.. driver. So I say lets keep this open and try to merge the functionality to the new driver.

**Comment by bapclenet** [comment]
@rousselk, do you think you will have time to adapt your changes to the new at86rf2xx driver? Or should someone else do it.

**Comment by OlegHahm** [comment]
@bapclenet, @daniel-k, @thomaseichinger, any volunteers to port these changes to the new driver?

**Comment by daniel-k** [comment]
I'm too busy right now. Maybe sometime later if these features proof useful to me, but I need to get other things done at the moment, sorry.

**Comment by bapclenet** [comment]
The same as daniel-k, I'm too busy at the moment, I hope I will be able to work on RIOT soon.

**Comment by rousselk** [comment]
I have just upgraded to current RIOT codebase (from 2014.12 release !).
This branch is now just irrelevant. I'm closing it, and will create a new branch/PR when ready.

**Comment by bapclenet** [comment]
Great! Thank you @rousselk

**Pull request closed without merging**